scripts/pages/home/model.mjs

```diff
@@ -0,0 +1,236 @@
+        } else {
+            profileInfoData = await profileInfoDataManager.fetch();
+        }
+
+        const profileWlDataManager = new DataManager(GET_PROFILE_WL_URL, { requestOptions: OPENDOTA_REQUEST_OPTIONS });
+
+        if (profileWlDataManager.cache) {
+            profileWlData = profileWlDataManager.cache;
+        } else {
+            profileWlData = await profileWlDataManager.fetch();
+        }
+
+        let profileInfo = {
+            personaname: profileInfoData.profile.personaname,
+            avatarfull: profileInfoData.profile.avatarfull,
+            wl: profileWlData,
+            winrate: (profileWlData.win / (profileWlData.win + profileWlData.lose) * 100).toFixed(2),
+        };
+
+        return profileInfo;
+    },
+
+    async getMatchesInfo() {
+        let matchesInfoData = [];
+
+        let heroesIdData = null;
+        let itemsIdData = null;
+
+        $.getJSON('json/heroes-id.json', function (result) {
+            heroesIdData = result.result.heroes;
+            //console.log(heroesIdData);
+        })
+
+        $.getJSON('json/items-id.json', function (result) {
+            itemsIdData = result;
+            console.log(itemsIdData);
+        })
+
+        const matchesDataManager = new DataManager(GET_MATCHES_HISTORY_URL, { params: RAPID_API_MATCHES_URL_PARAMS, requestOptions: RAPID_API_REQUEST_OPTIONS });
+
+        const matchesData = await matchesDataManager.fetch();
+        const matches = matchesData.result.matches;
+
+        console.log(matches);
+
+        if (matches.length === 0) {
+            throw new Error("No matches!");
+        }
+
+        const callbackQueue = [];
+
+        for (const match of matches) {
+            const params = { key: RAPID_API_MATCHES_URL_PARAMS.key, account_id: '76561198950833611', match_id: match.match_id };
+
+            const matchDetailDataManager = new DataManager(
+                GET_MATCH_DETAILS_URL,
+                { params, requestOptions: RAPID_API_REQUEST_OPTIONS }
+            );
+
+            if (matchDetailDataManager.cache) {
+                matchesInfoData.push(matchDetailDataManager.cache)
+            } else {
+                const fetchCallback = matchDetailDataManager.fetch.bind(matchDetailDataManager);
+                callbackQueue.push(fetchCallback);
+            }
+        }
+
+        if (callbackQueue.length > 0) {
+            let index = 0;
+            await new Promise((resolve, reject) => {
+                const intervalId = setInterval(() => {
+                    matchesInfoData.push(callbackQueue[index]());
+                    if (index >= callbackQueue.length - 1) {
+                        clearInterval(intervalId);
+                        return resolve();
+                    }
+                    index++;
+                }, 25);
+            })
+
+            matchesInfoData = await Promise.all(matchesInfoData);
+        }
+
+        let matchesArray = [];
+
+        for (const match of matchesInfoData) {
```

Comment:
I would extract the following code into a different method called `mapMatchesInfo`